### ISSUE TITLE
Allow $params to be passed ApiResource::refresh() method

### ISSUE THREAD

**Issue by onpubcom** [stripe/stripe-php#1035]
For example, ideally we could refresh an existing `Customer` object like:

```
$params = ['id' => $id, 'expand' => ['subscriptions']];
$customer = \Stripe\Customer::retrieve( $params );
$customer->subscriptions->create( $sub_params );
$customer->refresh(['expand' => ['subscriptions']]);
```

Alternatively an `ApiResource` object would "remember" what params it was originally passed and would just pass those same params to the Stripe API when `refresh()` is called.

Since especially in the case of the `Customer` object, the most recent Stripe API version no longer expands the `subscriptions` and `sources` collections by default, and as it stands now calling `refresh()` sets these collections back to null if they were previously expanded during initial retrieval, and there's no obvious way to workaround this with the SDK at the moment that I can see.

Looks like the `update()` method does allow `$params` to be passed, so perhaps `refresh()` can be updated to have a similar capability?

Thanks for your consideration. -Corey

PS: One other limitation of the latest SDK in the pseudo-code example above is the need to call `refresh()` at all in this particular case. Ideally, once the new Customer Subscription object is created, the SDK would just prepend it to the existing `$customer->subscriptions->data` array, but that is perhaps a separate issue altogether!

**Comment by remi-stripe** [comment]
@onpubcom Thanks for reaching out! While stripe-php has supported interacted with objects this way, we have decided to move away from this pattern earlier in the year. Instead, we shipped a client/service infrastructure that more explicitly separate the resource you have and how you interact with it. This is documented in our wiki [here](https://github.com/stripe/stripe-php/wiki/Migration-to-StripeClient-and-services-in-7.33.0).

The idea is that you wouldn't retrieve a customer to create a subscription and then refresh the customer object like you do int that code. The Subscription API for example has had a direct API `/v1/subscriptions` for multiple years and going through the customer itself is legacy. The code you have should be transformed to look like this instead: 

```
$stripe = new \Stripe\StripeClient('sk_test_xxxx');
$subscription = $stripe->subscriptions->create([
  'customer' => 'cus_123',
  ...
]);
$customer = $stripe->customers->retrieve(
  'cus_123',
  [
    'expand' => ['subscriptions'],
  ]
);
```

In the future, we will likely deprecate methods like `save()` or `refresh` on the Stripe objects and we do recommend moving to a client-service world in your integration.
